src/gate_detect/src/GateDetection.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: February 27th, 2018
+ * Description: Filter through image and determine if gate is in image and how
+ * far away it is
+ */
+
+#include "GateDetection.h"
+
+GateDetection::GateDetection(int cannyLow,
+                             int houghLinesThreshold,
+                             int houghLinesMinLength,
+                             int houghLinesMaxLineGap,
+                             int poleMax) {
+    _cannyLow = cannyLow;
+
+    _poleMax = poleMax;
+
+    _houghLinesThreshold = houghLinesThreshold;
+
+    _houghLinesMinLength = houghLinesMinLength;
+
+    _houghLinesMaxLineGap = houghLinesMaxLineGap;
+
+    _lowVertThresh = 20;
+
+    _lowHorThresh = 60;
+
+    m_Hor = m_Vert = 81.88;
+
+    b_Hor = b_Vert = -.92791;
+}
+
+GateDetection::GateDetection() {
+    _cannyLow = 50;
+
+    _poleMax = 200;
+
+    _houghLinesThreshold = 50;
+
+    _houghLinesMinLength = 50;
+
+    _houghLinesMaxLineGap = 50;
+
+    _lowVertThresh = 20;
+
+    _lowHorThresh = 60;
+
+    m_Hor = m_Vert = 81.88;
+
+    b_Hor = b_Vert = -.92791;
+}
```

Comment:
The interpolation constants seem a bit sketchy to put as a rosparam because they should be set during calibration and not changed, but we could potentially do that